huaweicloud/services/live/resource_huaweicloud_live_domain.go

```diff
@@ -94,132 +94,180 @@ func ResourceDomain() *schema.Resource {
+	)
+
+	client, err := cfg.NewServiceClient("live", region)
 	if err != nil {
-		return diag.Errorf("error creating Live v1 client: %s", err)
+		return diag.Errorf("error creating Live client: %s", err)
 	}
 
-	createOpts, err := buildCreateParams(d, region, c)
-	if err != nil {
-		return diag.FromErr(err)
+	createPath := client.Endpoint + httpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", client.ProjectID)
+	createOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         buildCreateDomainBodyParams(d, region),
 	}
 
-	_, err = client.CreateDomain(createOpts)
+	resp, err := client.Request("POST", createPath, &createOpt)
 	if err != nil {
 		return diag.Errorf("error creating Live domain: %s", err)
 	}
 
-	d.SetId(createOpts.Body.Domain)
-
-	err = waitingForDomainStatus(ctx, client, d.Id(), model.GetDecoupledLiveDomainInfoStatusEnum().ON,
-		d.Timeout(schema.TimeoutCreate))
+	respBody, err := utils.FlattenResponse(resp)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
-	// associate the streaming domain name with an ingest domain
-	err = associatingDomain(d, client)
+	domainName := utils.PathSearch("domain", respBody, "").(string)
+	if domainName == "" {
+		return diag.Errorf("error creating Live domain: domain name is not found in API response")
+	}
+
+	d.SetId(domainName)
+
+	err = waitingForDomainStateCompleted(ctx, client, d.Timeout(schema.TimeoutCreate), d.Id())
 	if err != nil {
-		return diag.FromErr(err)
+		return diag.Errorf("error waiting for the Live domain (%s) creation to complete: %s", d.Id(), err)
+	}
+
+	// Associate the streaming domain name with an ingest domain.
+	domainType := d.Get("type").(string)
+	if ingestDomain, ok := d.GetOk("ingest_domain_name"); ok && domainType == "pull" {
+		err = associateIngestDomain(client, d.Id(), ingestDomain.(string))
+		if err != nil {
+			return diag.FromErr(err)
+		}
 	}
 
-	// off the domain
-	if d.Get("status").(string) == "off" {
-		err = updateStatus(ctx, d, client, c)
+	// Disable the domain.
+	status := d.Get("status").(string)
+	if status == "off" {
+		err = updateDomainStatus(ctx, d, client, d.Timeout(schema.TimeoutCreate), d.Id(), status)
 		if err != nil {
 			return diag.FromErr(err)
 		}
 	}
 
+	// Enable the IPv6.
 	if d.Get("is_ipv6").(bool) {
-		if err := updateIPv6Switch(client, d); err != nil {
+		if err := updateIPv6Switch(client, d, d.Id()); err != nil {
 			return diag.Errorf("error updating Live domain IPv6 switch in creation operation: %s", err)
 		}
 	}
 
 	return resourceDomainRead(ctx, d, meta)
 }
 
+func buildCreateDomainBodyParams(d *schema.ResourceData, region string) map[string]interface{} {
+	params := map[string]interface{}{
+		"domain":                d.Get("name"),
+		"domain_type":           d.Get("type"),
+		"region":                region,
+		"service_area":          utils.ValueIgnoreEmpty(d.Get("service_area")),
+		"enterprise_project_id": utils.ValueIgnoreEmpty(d.Get("enterprise_project_id")),
```

Comment:
```suggestion
		"enterprise_project_id": utils.ValueIgnoreEmpty(d.Get("enterprise_project_id")),
```
Please use the common method